pkg/cmd/workspace/start.go

```diff
@@ -77,15 +78,25 @@ var StartCmd = &cobra.Command{
 			if err != nil {
 				log.Fatal(apiclient.HandleErrorResponse(res, err))
 			}
+
+			views.RenderInfoMessage(message)
+
+			if ideCodeFlag {
+				CodeCmd.Run(cmd, []string{workspaceId})
```

Comment:
I'm not fond of running sibling commands this way. I think we should extract a function from "code" which would be responsible for setting provider meta data and opening the IDE and reuse it here.
 
The current approach repeats all the commands above such as getting the config, apiclient and a "get" api call which we don't want to happen and also might result in maintenance issues if we do some UX changes for CodeCmd.